openmc/tallies.py

```diff
@@ -0,0 +1,3204 @@
+            if self.nuclides != other.nuclides:
+                return False
+
+        # Either all scores should match, or none should
+        return no_scores_match or all_scores_match
+
+    def can_merge(self, other):
+        """Determine if another tally can be merged with this one
+
+        If results have been loaded from a statepoint, then tallies are only
+        mergeable along one and only one of filter bins, nuclides or scores.
+
+        Parameters
+        ----------
+        other : openmc.Tally
+            Tally to check for merging
+
+        """
+
+        if not isinstance(other, Tally):
+            return False
+
+        # Must have same estimator
+        if self.estimator != other.estimator:
+            return False
+
+        equal_filters = sorted(self.filters) == sorted(other.filters)
+        equal_nuclides = sorted(self.nuclides) == sorted(other.nuclides)
+        equal_scores = sorted(self.scores) == sorted(other.scores)
+        equality = [equal_filters, equal_nuclides, equal_scores]
+
+        # If all filters, nuclides and scores match then tallies are mergeable
+        if equal_filters and equal_nuclides and equal_scores:
+            return True
+
+        # Variables to indicate matching filter bins, nuclides and scores
+        merge_filters = self._can_merge_filters(other)
+        merge_nuclides = self._can_merge_nuclides(other)
+        merge_scores = self._can_merge_scores(other)
+        mergeability = [merge_filters, merge_nuclides, merge_scores]
+
+        if not all(mergeability):
+            return False
+
+        # If the tally results have been read from the statepoint, we can only
+        # at least two of filters, nuclides and scores must match
+        elif self._results_read and sum(equality) < 2:
+            return False
+        else:
+            return True
+
+    def merge(self, other):
+        """Merge another tally with this one
+
+        If results have been loaded from a statepoint, then tallies are only
+        mergeable along one and only one of filter bins, nuclides or scores.
+
+        Parameters
+        ----------
+        other : openmc.Tally
+            Tally to merge with this one
+
+        Returns
+        -------
+        merged_tally : openmc.Tally
+            Merged tallies
+
+        """
+
+        if not self.can_merge(other):
+            msg = 'Unable to merge tally ID="{0}" with ' \
+                   '"{1}"'.format(other.id, self.id)
+            raise ValueError(msg)
+
+        # Create deep copy of tally to return as merged tally
+        merged_tally = copy.deepcopy(self)
+
+        # Differentiate Tally with a new auto-generated Tally ID
+        merged_tally.id = None
+
+        # Create deep copy of other tally to use for array concatenation
+        other_copy = copy.deepcopy(other)
+
+        # Identify if filters, nuclides and scores are mergeable and/or equal
+        merge_filters = self._can_merge_filters(other)
```

Comment:
`can_merge_filters` instead of `merge_filters`
`can_merge_nuclides` and `can_merge_scores` as well.